src/platform/amd/renoir/lib/clk.c

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+/*
+// Copyright(c) 2020 AMD.
+
+// Author: Basavaraj Hiregoudar<basavaraj.hiregoudar@amd.com>
+*/
+#include <sof/common.h>
+#include <sof/lib/clk.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+#include <platform/chip_registers.h>
+
+static struct freq_table platform_cpu_freq[] = {
+	{600000000, 600000 },
+};
+
+STATIC_ASSERT(NUM_CPU_FREQ == ARRAY_SIZE(platform_cpu_freq),
+				invalid_number_of_cpu_frequencies);
+
+static SHARED_DATA struct clock_info platform_clocks_info[NUM_CLOCKS];
+
+void  get_response_from_smu(void)
+{
+	uint32_t retValue = 0;
+	uint32_t count = 0;
+	do {
+		retValue = acp_reg_read_via_smn(MP1_SMN_C2PMSG_93, retValue, sizeof(int));
+		if (count > 0x007fffff) {
+			count = 0;
+		}
+		count++;
+	} while (0 == retValue);
+}
+
+/*********************************************************************************\
+* AcpChangeClockNotify()
+*/
+/**  \note      The \b AcpChangeCLock function details are provided in
+*               aos.h file.
+*
+***********************************************************************************/
+
+void acp_change_clock_notify (uint32_t	clock_freq)
+{
+	clock_freq = (clock_freq / 1000000);/*converting Hz to Mhz*/
+	/*Writing response zero to the response register so that it cab reset to '0' for further response*/
+	acp_reg_write_via_smn(MP1_SMN_C2PMSG_93,
+				0, sizeof(int));
+	/*Writing clock frequency value in Mhz in the Argument register*/
+	acp_reg_write_via_smn(MP1_SMN_C2PMSG_85,
+				clock_freq, sizeof(int));
+	/*Writing Message to message register*/
+	/*Writing Aclk change message to message register*/
+	acp_reg_write_via_smn(MP1_SMN_C2PMSG_69,
+				ACP_SMU_MSG_SET_ACLK, sizeof(int));
+	/*Waiting for SMU response*/
+	get_response_from_smu();
+}
+
+/*****************************************************************************\
+* acp_reg_read_via_smn()
+*/
+/**  \note      The \b acp_reg_read_via_smn function details are provided in
+*               acp_hal.h file.
+*
+*****************************************************************************/
+int acp_reg_read_via_smn (uint32_t	reg_offset,
+			 uint32_t	reg_value,
```

Comment:
why do you pass `reg_value` as a parameter? You only use it as a local variable. Was is supposed to be a pointer?